### ISSUE TITLE
subtyping check of pi type

### ISSUE THREAD

**Issue by yongweiy** [zhezhouzz/coreeff#2]
Got exception when checking subtyping between pi types.
```
Uncaught exception:
  
  (Failure "[file subtyping/sub.ml line 35]: die")
```

`let phi = (true : [%v: int -> bool]) in` in `let[@assert]` seems to be parsed as sigma type. What's the correct way of writing pi type in `let[@assert]`?

**Comment by zhezhouzz** [comment]
I think the Pi type is not completed implemented -- now it only works for the current benchmarks. I will working on it soon.

**Comment by yongweiy** [comment]
Makes sense. For the purpose of subtype checking, it seems that we can work around this by introducing pi type as additional arguments to `let[@assert]`.
